src/main/java/com/mael/ttt/ui/UserInterface.java

```diff
@@ -105,4 +108,12 @@ public boolean replay() {
         print(REPLAY);
         return console.read().equals("y");
     }
+
+    public List<String> getBoardIndexesAsStrings(int size) {
```

Comment:
That is true, yet it's among the same lines as feature envy, since you're replicating behaviour that is actually part of `Board` itself.
The list of valid indexes shouldn't necessarily leak into any client of `Board`. Think of `Board` as a black box. You give it a move/location and it's either rejected or accepted. `Board` itself can tell you whether it's out of range or already occupied, because `Board` knows all these things that are coherent to its responsibility.